### PR TITLE
Create connection pool to throw instead of return null

### PR DESCRIPTION
**Pull request by sgryphon**
This is one of the tasks part of getting ready for Extensions GA #95 

Note that an empty configuration is handled as Unknown and defaults to localhost; this situation covered by this change is where an unsupported connection pool type is used, or an unnamed enum value (e.g. 576).

If null is returned then subsequent code will store the value and try to call a property in the constructor of ConnectionConfiguration<T>, getting a less meaningful NullReferenceException.

### CODE REVIEW

File: src/Elastic.Ingest/ShipTo.cs [hunk 1]
```diff
@@ -85,9 +82,8 @@ public ShipTo(string cloudId, string username, string password)
 					var basicAuthCredentials = new BasicAuthenticationCredentials(Username, Password);
 					return new CloudConnectionPool(CloudId, basicAuthCredentials);
 				default:
-					return null;
+					throw new NotImplementedException($"Unrecognised connection pool type '{ConnectionPool}' specified in the configuration.");
```
Comment (russcam) [inline src/Elastic.Ingest/ShipTo.cs:85]: Maybe `ArgumentException` is a good fit for this?

**Review by russcam — APPROVED** [review]
LGTM @sgryphon, left a comment about the type of exception

File: src/Elastic.Ingest/ShipTo.cs (hunk 1 above)
Comment (sgryphon) [reply src/Elastic.Ingest/ShipTo.cs]: Changed.

**Comment by russcam** [comment]
@sgryphon seems like there might be a [missing `using` directive in src/Elastic.Ingest/ShipTo.cs](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-dotnet%2Fecs-dotnet-mbp/detail/PR-126/7/pipeline#step-189-log-301
)?

```
/var/lib/jenkins/workspace/ent-dotnet_ecs-dotnet-mbp_PR-126/src/go.elastic.co/apm/ecs-dotnet/src/Elastic.Ingest/ShipTo.cs(55,10): error CS0246: The type or namespace name 'IElasticLowLevelClient' could not be found (are you missing a using directive or an assembly reference?) [/var/lib/jenkins/workspace/ent-dotnet_ecs-dotnet-mbp_PR-126/src/go.elastic.co/apm/ecs-dotnet/src/Elastic.Ingest/Elastic.Ingest.csproj]
```

**Comment by sgryphon** [comment]
Merge issue... IElasticLowLevelClient had been replaced by ITransport<>, but I didn't remove the field on merge. Sorry, rushed the local checks too much. Should be good now.

**Comment by russcam** [comment]
@sgryphon No worries! I've just kicked off CI now. If all green, happy for this to be merged?

**Pull request merged**